multiqc/modules/base_module.py

```diff
@@ -18,6 +18,25 @@
 logger = logging.getLogger(__name__)
 
 
+class ModuleNoSamplesFound(Exception):
+    """Module checked all input files but couldn't find any data to use"""
+
+
+class ModuleBadInputError(Exception):
+    """Something with the input files, so we cannot proceed running the module"""
+
+    def __init__(self):
+        # Overriding the default init to explicitly forbid passing messages.
+        # We want users to print their own messages with log.error(), so it
+        # is gets automatically labelled with the module name in the log.
+        super().__init__()
```

Comment:
Would be nice if we could automate the logging. Would need to update the logger to be able to accept an optional namespace value, and use this over module name if supplied.

Low priority, so if not immediately easy to do we stick with this ☝🏻

Reply:
I'd actually stick to a for conventional use of the exception by passing a message:

```py
raise ModuleBadInputError("Mixed 'TOTAL_READS' and 'TOTAL_BASES' reports")
```

And then print the module name in `multiqc.py `:

```py
        except ModuleBadInputError as e:
            logger.error(f"{this_module}: could not run module: {e}")
```